### ISSUE TITLE
Bug in latest updates?

### ISSUE THREAD

**Issue by Alxandr** [genbtc/AutoTrimps#50]
I'm not entirely sure if this is caused by AutoTrimps, but there is an issue when I open the dropdown for autoportal lately. It throws my PC into a loop where it opens and closes the dropdown forcing me to do a force-reboot, and it's consistent. Has anything been changed with regards to the UI or similar the last few days?

**Comment by genbtc** [comment]
Thats weird. only that one?

**Comment by genbtc** [comment]
Weird. I see the GUI update function is being called like 30 times a second which is really bad and idk I will have to track it down. Strange but it works for me regardless.  

EDIT: OK i tracked it down and I see i have to rewrite the function so it doesnt go crazy.

**Comment by genbtc** [comment]
OK I did something to help fix it. 67d729ed97b2c6176e86f5e37004c4d86259b959
Btw what browser, OS, and speed of computer do you use? Even at 30 times a second, its not anything to me. I changed it down to 1 time a second.
Also - rebooting the entire computer? That seems crazy.

**Comment by Alxandr** [comment]
Yeah, I agree. It's OS-X, and a high power laptop. What I see happening though is that the dropdown keeps disappearing and reappearing, both of which are done while fading. So if I was to guess, I'd say that the issue is that when you remake the GUI (30 times/sec) while the dropdown is open, OS-X fades in and out the dropdown (which takes more than 1/30th of a second), so it's stuck in a loop of fading in and fading out system level UI, which causes all other UI to hang. I couldn't shut down chrome, I couldn't click any button anywhere :-/.

**Comment by genbtc** [comment]
Are you still having this issue with 1 update per second ?

OSX is an operating system not a browser (obviously) so how is a a webpage blocking "system level UI" ?  If what your saying is true = that Chrome GUI operations block the main OSX operating system during effects = that is a horrible and severe chrome+OSX bug. 

I see now there is a tag of "animation: fadeEffect 1s;" on the tab content window. But that shouldnt get inherited by the dropdowns. On Win10+Chrome it doesnt, and isnt meant to. 

It does seem to be a bug with OSX - theres enough documented problems on google when I google: "CSS animation fadeEffect OSX Chrome bug lag"

Use this website to test it, they should all be smooth (and use very low 0-1% CPU): https://daneden.github.io/animate.css/

If not, Try this fix out: https://david.gyttja.com/2013/01/21/fix-lagging-display-performance-on-retina-macbook-pro/
or this (seems similar but from apple)
https://discussions.apple.com/thread/6623697
It seems like its the Fade-Effect animation bug in OSX+Chrome that is severely bugged, and causing crippling low-performance on machines that should be able to handle it no problem.  See other reports of it on https://github.com/nwjs/nw.js/issues/3030 or https://github.com/nwjs/nw.js/issues/2723  This is a Chrome on OSX issue and this is very troubling and I cant test on OSX.

I am putting in a trick into the fade tag called: transform: translateZ(0); to enable hardware acceleration, as i read that can help. 7c75b12b9c3c4f88826d28673f9552110516e110

**Comment by Alxandr** [comment]
It's not unusable anymore, you can see what happens on the gif bellow. Also, I don't think this is related to css animations, cause I don't think you can control opening and closing animations of select options with css, but I might be wrong about that. If the gif is broken (too big or similar), I also uploaded the screen capture to youtube: https://youtu.be/FmO4etUv3d0

![autotrimps bug mov](https://cloud.githubusercontent.com/assets/112334/21172519/b2d0ddae-c1d2-11e6-85ca-0036f2e582bd.gif)

**Comment by Alxandr** [comment]
If you don't mind, Ima take a stab at completely overhauling the UI :)

**Comment by genbtc** [comment]
Go ahead. You think this dropdown fading thing is an OSX thing? If so that would explain your OS locking up. And would also explain whats causing the fading, if its not the animation tag I thought was being incorrectly inherited by the dropdown. So on OSX+Chrome its fading and flashing, on Win10+Firefox its just flashing(reselecting the old option) and on WIn10+Chrome it does neither.....

What i dont get is why it suddenly stopped fading and flashing. Is it from when you clicked the existing option ("Off") ? When that happened, thats how it looks on my end. Just normal.  

The update UI function being called is : updateCustomButtons()  in NEWUI2.js  
document.getElementById('AutoPortal').value = autoTrimpSettings.AutoPortal.selected;

If there is a better way to do it, by all means tell me

**Comment by Alxandr** [comment]
When I click any option it stops flashing. I just didn't want to pick anything bug "off" in case it accidentally immediately autoportalled. I'm looking at using Inferno for the UI and let it deal with the updates (it's like react, but much smaller) so that you don't update if there are no changes.

**Comment by genbtc** [comment]
adding a whole new framework sounds like a lot of work. now that I think about it, the 3 dropdowns that are updated in that function can be removed entirely from the function since they arent changed by the software, and moved into some new function that is called once on load, import or reset, but not update.

**Comment by Alxandr** [comment]
In general I would agree with you, but I'm at the same time looking at some general cleanup options.

![image](https://cloud.githubusercontent.com/assets/112334/21179923/a2fcd8aa-c1f6-11e6-8bed-96b83f7e094c.png)
